projects/valtimo/components/src/lib/components/form-io/form-io-address/form-io-address.component.ts

```diff
@@ -0,0 +1,191 @@
+        margin-top: 2px;
+    }
+  `]
+})
+export class FormIoAddressComponent implements FormioCustomComponent<any> {
+  @Input() value: any;
+  @Input() disabled: boolean = false;
+  @Input() mask: boolean = false;
+  @Input() required: boolean = false;
+  @Input() requiredStreet: boolean = false;
+  @Input() requiredHouseNumber: boolean = false;
+  @Input() requiredHouseLetter: boolean = false;
+  @Input() requiredHouseNumberAddition: boolean = false;
+  @Input() requiredPostalCode: boolean = false;
+  @Input() requiredCity: boolean = false;
+  @Input() keyLanguage: any;
+  @Output() valueChange = new EventEmitter<any>();
+
+  addressForm = new FormGroup({
+    street: new FormControl(''),
+    houseNumber: new FormControl(''),
+    houseLetter: new FormControl(''),
+    houseNumberAddition: new FormControl(''),
+    postalCode: new FormControl(''),
+    city: new FormControl('')
+  });
+
+  constructor() {
+    console.log('this: ', this);
+    setTimeout( () => {
+      console.log('langcon: ', this.keyLanguage);
+
+      this.addressForm.controls.street.setValue(this.value?.street);
+      this.addressForm.controls.street.setValidators(this.requiredStreet ? Validators.required : []);
+      this.addressForm.controls.street.updateValueAndValidity();
+
+      this.addressForm.controls.houseNumber.setValue(this.value?.houseNumber);
+      this.addressForm.controls.houseNumber.setValidators(this.requiredHouseNumber ? [Validators.required, Validators.pattern(/^[0-9]*[1-9][0-9]*$/)] : Validators.pattern(/^[0-9]*[1-9][0-9]*$/));
+      this.addressForm.controls.houseNumber.updateValueAndValidity();
+
+      this.addressForm.controls.houseLetter.setValue(this.value?.houseLetter);
+      this.addressForm.controls.houseLetter.setValidators(this.requiredHouseLetter ? Validators.required : []);
+      this.addressForm.controls.houseLetter.updateValueAndValidity();
+
+      this.addressForm.controls.houseNumberAddition.setValue(this.value?.houseNumberAddition);
+      this.addressForm.controls.houseNumberAddition.setValidators(this.requiredHouseNumberAddition ? Validators.required : []);
+      this.addressForm.controls.houseNumberAddition.updateValueAndValidity();
+
+      this.addressForm.controls.postalCode.setValue(this.value?.postalCode);
+      this.addressForm.controls.postalCode.setValidators(this.requiredPostalCode ? [Validators.required, Validators.pattern(/^[0-9]{4} ?[A-Z]{2}$/)] : Validators.pattern(/^[0-9]{4} ?[A-Z]{2}$/));
+      this.addressForm.controls.postalCode.updateValueAndValidity();
+
+      this.addressForm.controls.city.setValue(this.value?.city);
+      this.addressForm.controls.city.setValidators(this.requiredCity ? Validators.required : []);
+      this.addressForm.controls.city.updateValueAndValidity();
+
+      if (this.disabled) {
+        Object.keys(this.addressForm.controls).forEach((key) => {
+          this.addressForm.get(key).disable();
+        })
+      }
+
+      this.addressForm.valueChanges.subscribe(() => {
```

Comment:
subscription is opened but never closes. Please look in this PR how subscriptions should be handled: https://github.com/valtimo-platform/valtimo-frontend-libraries/pull/938/files